### ISSUE TITLE
EMNIST evaluation does not evaluate last 7 letters

### ISSUE THREAD

**Issue by lynxloco** [deeplearning4j/deeplearning4j#7732]
#### Issue Description

When testing the EMNIST dataset, the final 7 classes (TUVWXYZ) are not being evaluated and gives the following warning: Warning: 7 classes were never predicted by the model and were excluded from average recall
Classes excluded from average recall: [19, 20, 21, 22, 23, 24, 25]

Please describe our issue, along with:
- expected behavior
All classes are evaluated.
- encountered behavior
Not all classes are evaluated.

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version
1.0.0-bet4
* platform information (OS, etc)
Ubuntu 18.04
* CUDA version, if used
10.0
* NVIDIA driver version, if in use
418

#### Contributing

If you'd like to help us fix the issue by contributing some code, but would
like guidance or help in doing so, please mention it!
```
public static void main(String[] args) throws Exception {
        int height = 28;    // height of the picture in px
        int width = 28;     // width of the picture in px
        int channels = 1;   // single channel for grayscale images
        int outputNum = 26; // 26 character classification
        int batchSize = 54; // number of samples that will be propagated through the network in each iteration
        int nEpochs = 1;    // number of training epochs

        int seed = 1234;    // number used to initialize a pseudorandom number generator.
        Random randNumGen = new Random(seed);

        EmnistDataSetIterator.Set emnistSet = EmnistDataSetIterator.Set.LETTERS;
        EmnistDataSetIterator trainIter = new EmnistDataSetIterator(emnistSet, batchSize, true, randNumGen.nextInt());
        EmnistDataSetIterator testIter = new EmnistDataSetIterator(emnistSet, batchSize, false, randNumGen.nextInt());

        LOGGER.info("Network configuration and training...");
        // reduce the learning rate as the number of training epochs increases
        // iteration #, learning rate
        Map<Integer, Double> learningRateSchedule = new HashMap<>();
        learningRateSchedule.put(0, 0.06);
        learningRateSchedule.put(200, 0.05);
        learningRateSchedule.put(600, 0.028);
        learningRateSchedule.put(800, 0.0060);
        learningRateSchedule.put(1000, 0.001);

        MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(seed)
                .l2(0.0005) // ridge regression value
                .updater(new Nesterovs(new MapSchedule(ScheduleType.ITERATION, learningRateSchedule)))
                .weightInit(WeightInit.XAVIER)
                .list()
                .layer(new ConvolutionLayer.Builder(5, 5)
                        .nIn(channels)
                        .stride(1, 1)
                        .nOut(26)
                        .activation(Activation.IDENTITY)
                        .build())
                .layer(new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
                        .kernelSize(2, 2)
                        .stride(2, 2)
                        .build())
                .layer(new ConvolutionLayer.Builder(5, 5)
                        .stride(1, 1) // nIn need not specified in later layers
                        .nOut(50)
                        .activation(Activation.IDENTITY)
                        .build())
                .layer(new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX)
                        .kernelSize(2, 2)
                        .stride(2, 2)
                        .build())
                .layer(new DenseLayer.Builder().activation(Activation.RELU)
                        .nOut(500)
                        .build())
                .layer(new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                        .nOut(outputNum)
                        .activation(Activation.SOFTMAX)
                        .build())
                .setInputType(InputType.convolutionalFlat(height, width, channels)) // InputType.convolutional for normal image
                .build();

        MultiLayerNetwork net = new MultiLayerNetwork(conf);
        net.init();
        net.setListeners(new ScoreIterationListener(10));
        LOGGER.info("Total num of params: {}", net.numParams());

        // evaluation while training (the score should go down)
        for (int i = 0; i < nEpochs; i++) {
            net.fit(trainIter);
            LOGGER.info("Completed epoch {}", i);
            Evaluation eval = net.evaluate(testIter);
            LOGGER.info(eval.stats(false, true));

            trainIter.reset();
            testIter.reset();
        }

        new File(BASE_PATH).mkdirs();
        File ministModelPath = new File(BASE_PATH + "/eminist-model.zip");
        ModelSerializer.writeModel(net, ministModelPath, true);
        LOGGER.info("The EMINIST model has been saved in {}", ministModelPath.getPath());
    }
}

**Comment by AlexDBlack** [comment]
> Warning: 7 classes were never predicted by the model and were excluded from average recall

This is not a bug or issue with evaluation. The error says "were never predicted by the model" - this means that the model isn't making good predictions. It's a network tuning issue, in other words - for example, a bad learning rate, not enough epochs, too small a network, etc.

Maybe start with these:
https://deeplearning4j.org/docs/latest/deeplearning4j-nn-visualization
https://deeplearning4j.org/docs/latest/deeplearning4j-troubleshooting-training

**Comment by lynxloco** [comment]
The network is copied directly from the mnist tutorial. We fixed it by creating our own Emnist Dataset iterator.

**Comment by raver119** [comment]
LeNet model you've copied - is suited for recognition of handwritten characters. You can't copy model as is and hope it'll work for everything else out there. There will be always some tweaks to be done.
